f5_openstack_agent/lbaasv2/drivers/bigip/sip_profile.py

```diff
@@ -0,0 +1,102 @@
+# coding=utf-8
+# Copyright (c) 2016-2023, F5 Networks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from f5_openstack_agent.lbaasv2.drivers.bigip import resource_helper
+from oslo_log import log as logging
+
+LOG = logging.getLogger(__name__)
+
+
+class SIPProfileHelper(object):
+    """A tool class for SIP profile process"""
+
+    def __init__(self):
+        self.sip_helper = resource_helper.BigIPResourceHelper(
+            resource_helper.ResourceType.sip_profile
+        )
+
+    @staticmethod
+    def enable_sip(service):
+        listener = service.get('listener')
+        if listener and listener.get('protocol') == 'SIP':
+            return True
+        return False
+
+    def add_profile(self, service, vip, bigip):
+        partition = vip['partition']
```

Comment:
Is it a vip or a vs in bigip?

Reply:
naming issue.  the repo had these names interchangeable.  in future, we might want to modify these names consistently in other places as well,  in ftp, tcp profile  etc.